plugins/org.obeonetwork.m2doc/src/org/obeonetwork/m2doc/generator/TemplateProcessor.java

```diff
@@ -262,36 +262,62 @@ private XWPFRun insertFieldRunReplacement(XWPFRun srcRun, String replacement) {
             createNewParagraph(srcRun.getParagraph());
             forceNewParagraph = false;
         }
-        // creates as many paragraphs as there are '\n's in the string.
-        String[] fragments = replacement.split("\n");
-        XWPFRun result = null;
-        int size = fragments.length;
-        for (int i = 0; i < size - 1; i++) {
-            XWPFRun generatedRun = insertFragment(fragments[i], srcRun);
-            if (result == null) {
-                result = generatedRun;
-            }
-            createNewParagraph(srcRun.getParagraph());
-        }
-        if (size > 0) {
-            XWPFRun generatedRun = insertFragment(fragments[size - 1], srcRun);
-            if (result == null) {
-                result = generatedRun;
+        return insertString(srcRun, replacement);
+    }
+
+    /**
+     * Insert the given text into to given {@link XWPFRun}. This take care of new lines and tabulations.
+     * 
+     * @param srcRun
+     *            the {@link XWPFRun} to populate
+     * @param text
+     *            the text
+     * @return the last inserted {@link XWPFRun}.
+     */
+    private XWPFRun insertString(XWPFRun srcRun, String text) {
+        int fragmentStart = 0;
+
+        XWPFRun inserted;
+        for (int i = 0; i < text.length(); i++) {
+            final char current = text.charAt(i);
+            switch (current) {
+                case '\n':
+                    inserted = insertFragment(srcRun, text.substring(fragmentStart, i));
+                    inserted.addBreak();
+                    fragmentStart = i + 1;
+                    break;
+                case '\t':
+                    inserted = insertFragment(srcRun, text.substring(fragmentStart, i));
+                    inserted.addTab();
+                    fragmentStart = i + 1;
+                    break;
+                case '\r':

```

Comment:
If I understand well, in case of a \r\n you add a break and in case of a \r, you just consumme the \r. 

Why don't you just consumme the '\r' whatever the next char is and leave to the for loop it's responsibility to increment the i variable and to the case '\n' it's responsibility to process the '\n'?

It seems to me that this would work the same way you're code is working and would be simpler to read.